flow/project.py

```diff
@@ -601,7 +601,7 @@ def __init__(self, config=None, environment=None):
         except KeyError:
             self._use_buffered_mode = False
 
-    def _setup_template_environment(self):
+    def _setup_template_environment(self, environment=None):
```

Comment:
Could you help me understand this change?

Reply:
Thanks for pointing this out, I was meaning to check with you on this. I need to add this because I notice that for the unittest as well as any mock submit test where self.env and the intended submit environment is not the same, the `_setup_template_environment function` currently won't have the submit environment but it is now needed for `template_environment.filters['get_prefix'] = environment.get_prefix` to get the correct get_prefix function from the actual submit env. I think the change should not break anything but would like your thoughts on this.

Reply:
The `FlowProject` instance already has the environment as `self._environment`:
https://github.com/glotzerlab/signac-flow/blob/893017073a9ce0203e03efbdcab75f094e2d6793/flow/project.py#L588 Is this not sufficient?

Reply:
I think the self._environment is set in the initialization of the FlowProject class but when we run submit() we currently allow user input of a different environment:
``` 
def submit(self, bundle_size=1, jobs=None, names=None, num=None, 
parallel=False, force=False, walltime=None, env=None, **kwargs):
..........
if env is None:
    env = self._environment
```
So they may not be the same and we need the environment provided in submit() to provide the correct filter function. I think this is also the way current unit test is setup so without the environment argument from submit only the base ComputeEnvironment.get_mpi will be used.

Reply:
Got it.

Reply:
With the PR you added this is no longer needed as template related filter is all added before this function all. Thanks for your help!

Reply:
I think this parameter can now be removed because the template filters are added in `_template_environment` instead.